status/data_deliveries_plot.py

```diff
@@ -0,0 +1,58 @@
+import json
+import datetime
+from status.util import dthandler, SafeHandler
+
+
+class DataDeliveryHandler(SafeHandler):
+    """ Handles the api call to proj_staged data
+    Loaded through /api/v1/proj_staged/([^/]*)$
+    """
+    def get(self, search_string=None):
+        staged_files_view = self.application.projects_db.view("project/staged_files_irma")
+        summary_view = self.application.projects_db.view("project/summary")
+        delivered_data = {}
+        if not search_string:
+            last_month = datetime.datetime.now()-datetime.timedelta(days=30)
+            first_term = last_month.isoformat()[:10]
+            second_term = datetime.datetime.now().isoformat()[:10]
+            search_string = "{}--{}".format(first_term, second_term)
+        searches = search_string.split('--')
+        for row in staged_files_view.rows:
+            filesize = 0
+            for key,val in row.value.items():
+                for key2,val2 in val.items():
+                    filesize += int(val2['size_in_bytes'])
+            project_name = row.key[0]
+            project_id_staged = row.key[1]
+            for row in summary_view.rows:
+                project_id_summary = row.key[1]
+                if project_id_staged == project_id_summary: 
+                    if 'project_summary' in row.value and 'details' in row.value  and 'all_raw_data_delivered' \
```

Comment:
I don't understand why you have all these cases in the 'if' but then also have the try-except case down below. They should serve the same purpose right?

Reply:
Because it gave me a `KeyError` previously, but of course now that I removed it, it worked! :-D

Reply:
Right, and also, which projects are removed by this if-filter? Are there any projects which lack these fields that should be included? I have no idea.

Reply:
There will definitely be some projects that are not included, I don't think there should be that many though? I think the plot should give a rough estimate of the data delivered visually. But maybe I could get a more accurate number for the total sum in the cumulative view by editing the back-end, I'll look into it!

Reply:
Yeah missing some projects might be fine, but we would like to know how many we're missing and why...  I'm not sure checking for all of these keys are necessary. But I might be wrong

Reply:
I added the projects that don't have all fields to the dictionary with their file size and date delivered as these are required. It didn't change the data delivered value that much though. I think it's because most projects that don't have the other fields filled in don't have a date delivered...

Reply:
Updated code with Johannes new view 🎉 

I figured out a way to include the project id's again, by adding a list of the id's to each project in the dictionary. This doesn't change the system that Johannes created to sync the axis. Though the synchronization doesn't work for me. Since it didn't work, I added the project names again to the hover pop-up :-)

The file size counter stopped working because of a find- and replace typo 😊 it works again!